src/ruby/lib/grpc/generic/active_call.rb

```diff
@@ -187,7 +201,7 @@ def finished
     # @param marshalled [false, true] indicates if the object is already
     # marshalled.
     def remote_send(req, marshalled = false)
-      # TODO(murgatroid99): ensure metadata was sent
+      send_initial_metadata unless @metadata_sent
```

Comment:
Since we check `@metadata_sent` under the mutex, I don't think it makes sense to check it again here or in other places where `send_initial_metadata` is called. It's a minor optimization, and it can still race.